galaxy_ng/app/api/v1/tasks.py

```diff
@@ -0,0 +1,254 @@
+    if not github_reference:
+        github_reference = None
+
+    role_name = alternate_role_name or github_repo.replace('ansible-role-', '')
+    if LegacyNamespace.objects.filter(name=github_user).count() == 0:
+        logger.info(f'CREATE NEW NAMESPACE {github_user}')
+        namespace, _ = LegacyNamespace.objects.get_or_create(name=github_user)
+    else:
+        logger.info(f'USE EXISTING NAMESPACE {github_user}')
+        namespace = LegacyNamespace.objects.filter(name=github_user).first()
+
+    with tempfile.TemporaryDirectory() as checkout_path:
+
+        clone_url = f'https://github.com/{github_user}/{github_repo}'
+        gitrepo = Repo.clone_from(clone_url, checkout_path, multi_options=["--recurse-submodules"])
+        github_commit = None
+        github_commit_date = None
+
+        if github_reference is not None:
+            gitrepo.checkout(github_reference)
+            github_commit = get_tag_commit_hash(
+                clone_url,
+                github_reference,
+                checkout_path=checkout_path
+            )
+            github_commit_date = get_tag_commit_date(
+                clone_url,
+                github_reference,
+                checkout_path=checkout_path
+            )
+        else:
+            # FIXME - figure out the old logic.
+            if gitrepo.tags:
+                tag = gitrepo.tags[-1]
+                github_reference = tag.name
+                github_commit = tag.commit.hexsha
+                github_commit_date = datetime.datetime.fromtimestamp(tag.commit.committed_date)
+                github_commit_date = github_commit_date.isoformat()
+
+        logger.info(f'GITHUB_REFERENCE: {github_reference}')
+        logger.info(f'GITHUB_COMMIT: {github_commit}')
+
+        # check if this namespace/name/version has already been imported
+        old = LegacyRole.objects.filter(namespace=namespace, name=role_name).first()
+        if old is not None:
+            old_versions = old.full_metadata.get('versions', [])
+            old_versions = [x['name'] for x in old_versions]
+            logger.info(f'OLD VERSIONS: {old_versions}')
+            if github_reference in old_versions:
+                msg = (
+                    f'{namespace.name}.{role_name} {github_reference}'
+                    + 'has already been imported'
+                )
+                raise Exception(msg)
+
+        role_meta = roles_utils.get_path_role_meta(checkout_path)
+        role_tags = role_meta.get('galaxy_info', {}).get('galaxy_tags', [])
+        if role_tags is None:
+            role_tags = []
+        logger.info(f'TAGS: {role_tags}')
+
+        # use the importer to grok the readme
+        ldr = RoleLoader(
+            content_type='role',
+            root=os.path.dirname(checkout_path),
+            rel_path=os.path.basename(checkout_path)
+        )
+        readme = ldr._get_readme()
+        readme_html = markup_utils._render_from_markdown(readme)
```

Comment:
This would prevent use of the private/internally used methods.
```suggestion
        readme = markup_utils.get_readme_doc_file(checkout_path)
        if not readme:
            raise Exception("No role readme found.")
        readme_html = markup_utils.get_html(readme)
```

Reply:
done, ty